### ISSUE TITLE
[RadioButton && Label]: [While navigating around most of the button & radio button within ‘XX’ dialogs, names & group names are not associated with the respective UIs.]

### ISSUE THREAD

**Issue by sanjayC5373903** [SAP/ui5-webcomponents-react#6112]
### Describe the bug

1. While navigating around most of the radio button within ‘XX’ dialogs, group names are not associated with the respective UIs.
![Screenshot 2024-07-24 102107](https://github.com/user-attachments/assets/8a1912c1-700f-4a18-bff6-19fe81cd0632)

<FlexBox direction="Column">
          <DialogFormField
            label={t('REPLENISHMENTS_STRATEGY')}
            accessibleName={t('demandDriven')}
            name="strategyGroup"
            as={RadioButton}
            direction={'Column'}
            text={t('REPLENISHMENT_STRATEGY_DEMAND_DRIVEN')}
            checked={checkDemandDriven(selected)}
            onClick={e => {
              setSelected('demandDriven');
              setValue('strategyGroup', 'demandDriven');
            }}
            disabled={checkFormView(formType)}
            data-testid="demandDriven"
          />
          <DialogFormField
            accessibleName={t('costOptimalDemandDriven')}
            name="strategyGroup"
            as={RadioButton}
            direction={'Column'}
            text={t('REPLENISHMENT_STRATEGY_COST_OPTIMAL_DEMAND_DRIVEN')}
            onClick={onClickRadioButton}
            checked={selected === 'costOptimalDemandDriven' ? true : false}
            disabled={checkFormView(formType)}
            data-testid="costOptimalDemandDriven"
          />
</Flexbox>

DialogFormField component return part below.

<S.DialogFormFieldFlexBox direction={direction}>
      {label &&
        (labelStyle ? (
          <S.Label required={required} direction={direction} showColon={colon} style={labelStyle}>
            {label}
          </S.Label>
        ) : (
          <S.Label required={required} direction={direction} showColon={colon}>
            {label}
          </S.Label>
        ))}
      {FormElement && (
        <FlexBox alignItems="center">
          <FormField {...props} name={name} as={FormElement} required={required} />
        </FlexBox>
      )}
    </S.DialogFormFieldFlexBox>

2. While navigating around most of the buttons within ‘XX’ dialogs, names are not associated with the respective UIs.
![Screenshot 2024-07-24 102045](https://github.com/user-attachments/assets/ad17596e-4f4b-40cd-9b8f-d5a9302d620b)

<FlexBox
            justifyContent="SpaceBetween"
            alignItems="Baseline"
            style={{ marginBottom: '0.5rem' }}
          >
            <Label required style={{ margin: '0.1875rem 0' }} showColon={true}>
              {t('SHELF_LIFE_RANGE')}
            </Label>
            <FlexBox justifyContent="End">
              <Button
                data-testid="add-button"
                design="Transparent"
                disabled={isAddDisabled()}
                onClick={async () => {
                  clearErrors('shelfLifeProfileRanges');
                  prepend({ fromdays: null, todays: null });
                  trigger('shelfLifeProfileRanges');
                }}
              >
                {t('ADD_RANGE')}
              </Button>
            </FlexBox>
          </FlexBox>

### Reproduction steps

1.
2.
3.
...

### Expected Behaviour

Every name and their respective group names should be associated together and should be read by the screen reader.

### UI5 Web Components for React Version

^1.16.3

### UI5 Web Components Version

^1.14.0

### Browser

Chrome

### Operating System

Windows11

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by sanjayC5373903** [comment]
Hi @MarcusNotheis Thanks for response.
Example  https://stackblitz.com/edit/github-i6cvg6?file=src%2FApp.tsx,src%2FRadioBtnGrpFile.tsx
 
As above url of code has below version and its working fine, its read the respective name by reader.
"@ui5/webcomponents": "~1.24.0", & "@ui5/webcomponents-react": "~1.29.0"
 
Could you please try with below version? I have tried to downgrade version. but its not working. So could you please check with this version and inform me?
"@ui5/webcomponents": "^1.14.0", & "@ui5/webcomponents-react": "^1.16.3"

**Comment by MarcusNotheis** [comment]
Hi @sanjayC5373903,

as I'm working on a Mac I can't reproduce the issue using JAWS, but here's the sandbox using the downgraded versions: https://stackblitz.com/edit/github-i6cvg6-gohsff?file=src%2FApp.tsx,package.json

Anyhow, as the issue seems to be fixed in the latest versions, I would strongly recommend to update to the latest 1.x release.
The versions you're trying to reproduce the issue on are already more than one year old.

**Comment by sanjayC5373903** [comment]
Hi @MarcusNotheis I have tested as given using the downgraded versions. but its not working as expectation. we need read label. and also name of input field like "edit combo" and "edit input" .

**Comment by MarcusNotheis** [comment]
Does the issue occur on the latest v1 versions (@ui5/webcomponents: 1.24.8, @ui5/webcomponents-react: 1.29.7) as well? As mentioned before, the versions you are running on are more than a year old, so we won't downport changes to that version. You should be able to easily upgrade to the latest v1 versions as there are no breaking changes included.
